### ISSUE TITLE
Transferring to train set was not going well

### ISSUE THREAD

**Issue by amadeusuzx** [lordmartian/deep_avsr#37]
Hi there,

I have been using your code to experiment on features extracted using a custom visual front. The pretraining was not bad, and on the validation set the WER reached about 83.1% while the CER was 36.3% after the curriculum learning. (1,2,3,5,7,9,13,17,21,29,37). 
```Step: 199 || Tr.Loss: 0.428430  Val.Loss: 2.574524 || Tr.CER: 0.109  Val.CER: 0.525 || Tr.WER: 0.363  Val.WER: 0.831```
(The step size was set 200 because I just wanted a quick try.)

However, the training phase was not so good, and it seems the pretrained model was not working because the train CER started from 95.5%.

Train_Val_Loss 
![Train_Val_Loss](https://user-images.githubusercontent.com/44519021/124113569-8d996a00-daa6-11eb-9f08-49b5ea2ddcc7.png)
Train_Val_WER
![Train_Val_WER](https://user-images.githubusercontent.com/44519021/124113575-8e320080-daa6-11eb-9d24-88bf334d59e7.png)　
Train_Val_CER　
![Train_Val_CER](https://user-images.githubusercontent.com/44519021/124113578-8eca9700-daa6-11eb-825d-aabd7721a419.png)

As you can see, the val CER reached about 50% at last while the WER remains 109.5%...
In your experiment, how did the CER/WER change when transferring to the training phase ? It will be really helpful if you can give me some advice.

**Comment by amadeusuzx** [comment]
I decoded on the test set with LM, and here are some results: 
```
Prediction: LAEVE BE LISH
Ground Truch: WHATEVER IT WAS
Prediction: WEE IS I FA I A FO SODE
Ground Truch: IN TERMS OF FIRE SERVICE RESOURCES
Prediction: MOS WE E E OE O O E A I I E REBMOT
Ground Truch: BUT WE'RE GOING TO IMPROVE ON THINGS LIKE THE GREEN ROOF
Prediction: IT FARY MO TETINS
Ground Truch: THE FAMILY SEAT
```
Do you have any suggestions?

**Comment by lordmartian** [comment]
Hi,
I didn't observe such behaviour. CER/WER starting from a higher value is normal but it should reduce as the training progresses. Are you doing anything else apart from loading the pretrained weights (for eg. freezing any weights, etc) before starting the training? Also, have you tried finetuning the hyperparameter values?

**Comment by amadeusuzx** [comment]
Thanks for your response!
I don't think I did anything like freezing model parts... Recently I also tried training on features extracted by the preprocessing.py instead of my method, however the results were quite similar, which converged at CER 65% but WER over 100% on training set.

**Comment by amadeusuzx** [comment]
Sorry, it looks like I forgot mention that I'm working on a video-only task.

**Comment by lordmartian** [comment]
Can you try once by changing the seed value or lowering the learning rate?

**Comment by amadeusuzx** [comment]
Certainly! I'll report here if there are any updates.

**Comment by amadeusuzx** [comment]
Sadly I still can't reproduce the WER 50% results... This time I used features extracted by preprocessing.py, and ran through the pretrain and train steps. I also changed the random seed value. Here are the curves:

****pretrain****
![image](https://user-images.githubusercontent.com/44519021/124632077-f9247280-debe-11eb-82b4-598d7a86fd75.png)
****train****
![image](https://user-images.githubusercontent.com/44519021/124632192-122d2380-debf-11eb-9b1b-4e26514955fb.png)

val WER was over 100% in the whole training phase. 
Do you have any comments about this?

**Comment by lordmartian** [comment]
Thanks for the update.

I do not have any exact answer for why is this happening or how to solve it as I hadn't faced such issue. From the graphs, it looks like the model is overfitting during the training phase. I will think about it but I am not sure I will find any exact solution. For now, you can try to finetune the hyperparameters to avoid the model from overfitting. By the way, are there any changes in the code or dataset that may be impacting the training?

Also, you need not rerun the pretraining phase every time as the WER after pretraining is close to what is expected. Just need to experiment with the training phase.

Regarding CER values, I do not have that data anymore.

**Comment by amadeusuzx** [comment]
Thanks for your reply. I think the only worth mentioning change is that I put the pretrain step in a loop so that I just need to ran once... Anyway, thanks again for your advice and I will try tune the parameters further!
